packages/antd/src/editable/index.tsx

```diff
@@ -31,7 +31,7 @@ const useEditable = (): [boolean, (payload: boolean) => void] => {
     if (pattern === 'editable') {
       return field.setPattern('readPretty')
     }
-  }, [pattern])
+  }, [pattern, field])
```

Comment:
to fix after sort, Editable Input editable state not re init.
![2023-06-15 11 38 14](https://github.com/alibaba/formily/assets/18055018/048c724c-5125-4924-87c5-2543ae837e42)

Reply:
fixed.